src/app/meme-card/meme-card.component.html

```diff
@@ -1,18 +1,19 @@
 <mat-card class="meme-card">
-  <img class="meme-img" [style.height.px]="checkHeight(imageHeight)" [style.minWidth.px]="minCardWidth(imageHeight)"
-       [style.maxWidth.px]="maxCardWidth(imageHeight)" [src]="image">
+  <img (click)="dialogPage()" class="meme-img" [style.height.px]="checkHeight(imageHeight)"
+       [style.minWidth.px]="minCardWidth()" [style.maxWidth.px]="maxCardWidth(imageHeight)" [src]="imageLink">
   <!-- Actions that are appended to the bottom of the card: show author name, up vote, down vote-->
   <mat-card-content class="actions" style="margin-top: -3px;">
     <span class="author">
       <a href="/">{{username}}</a>
     </span>
-
     <div class="cardActionButtons">
       <!-- Up vote button -->
-      <button mat-icon-button (click)="onClickUpVote()" [color]="voted == 1 ? 'primary': 'basic'"><i class="material-icons">thumb_up</i></button>
-      <span>{{voteCount}}</span>
+      <button mat-icon-button (click)="onClickUpVote()" [color]="_myVote == 1 ? 'primary': 'basic'"><i
+        class="material-icons">thumb_up</i></button>
+      <span>{{_totalVote}}</span>
       <!-- Down vote button -->
-      <button mat-icon-button (click)="onClickDownVote()" [color]="voted == -1 ? 'warn': 'basic'"><i class="material-icons">thumb_down</i></button>
+      <button mat-icon-button (click)="onClickDownVote()" [color]="_myVote == -1 ? 'warn': 'basic'"><i
```

Comment:
_myVote should be just myVote in the meme-card-component.

Reply:
Nevermind, the meme-card-component is changed in yours. lol... I'm so confused

Reply:
myVote was encapsulated with a setter method (to update the dialog if applicable) so the variable itself is called "_myVote" and the setter method is "myVote"